### PR TITLE
[armadillo] no absolute paths

### PR DESCRIPTION
**Pull request by autoantwort**
For #20469

### CODE REVIEW

**Review by LilyWangLL — APPROVED** [review]

File: ports/armadillo/portfile.cmake [hunk 1]
```diff
@@ -41,8 +41,11 @@ if(SHARE_LEN EQUAL 0)
     file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/share/Armadillo")
 endif()
 
-if (VCPKG_TARGET_IS_WINDOWS)
-    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/armadillo_bits/config.hpp" "#define ARMA_AUX_LIBS ${CURRENT_INSTALLED_DIR}/lib/openblas.lib;${CURRENT_INSTALLED_DIR}/lib/lapack.lib;${CURRENT_INSTALLED_DIR}/lib/openblas.lib" "")
+set(filename "${CURRENT_PACKAGES_DIR}/include/armadillo_bits/config.hpp")
+if(EXISTS "${filename}")
+    file(READ "${filename}" contents)
+    string(REGEX REPLACE "\n#define ARMA_AUX_LIBS .*\n" "\n" contents "${contents}")
```
Comment (ras0219-msft) [inline ports/armadillo/portfile.cmake:47]: Does this regex match multiple lines? Would it be simpler to just replace `#define ARMA_AUX_LIBS` with `//`?

Comment (autoantwort) [reply ports/armadillo/portfile.cmake:47]: It matches a single line. 

> Would it be simpler to just replace `#define ARMA_AUX_LIBS` with `//`?

Probably, depending on https://github.com/microsoft/vcpkg-tool/pull/172#discussion_r776026107

Comment (dg0yt) [reply ports/armadillo/portfile.cmake:47]: > It matches a single line.

`REGEX` matches only once, but `.*` also matches line breaks. So the proposed change matches multiple lines.

Comment (LilyWangLL) [reply ports/armadillo/portfile.cmake:47]: I tested this code locally when I reviewed this PR, and the result was as below.
Original result:
```
// if Armadillo was installed on this system via CMake and ARMA_USE_WRAPPER is not defined,
// ARMA_AUX_LIBS lists the libraries required by Armadillo on this system, and
// ARMA_AUX_INCDIRS lists the include directories required by Armadillo on this system.
// Do not use these unless you know what you are doing.
#define ARMA_AUX_LIBS H:/Lily/1210/vcpkg/installed/x64-windows/lib/openblas.lib;H:/Lily/1210/vcpkg/installed/x64-windows/lib/lapack.lib;H:/Lily/1210/vcpkg/installed/x64-windows/lib/openblas.lib
#define ARMA_AUX_INCDIRS 

```
The result before this PR:
```
// if Armadillo was installed on this system via CMake and ARMA_USE_WRAPPER is not defined,
// ARMA_AUX_LIBS lists the libraries required by Armadillo on this system, and
// ARMA_AUX_INCDIRS lists the include directories required by Armadillo on this system.
// Do not use these unless you know what you are doing.

#define ARMA_AUX_INCDIRS 

```
The rersult of this PR:
```
// if Armadillo was installed on this system via CMake and ARMA_USE_WRAPPER is not defined,
// ARMA_AUX_LIBS lists the libraries required by Armadillo on this system, and
// ARMA_AUX_INCDIRS lists the include directories required by Armadillo on this system.
// Do not use these unless you know what you are doing.

```

Comment (autoantwort) [reply ports/armadillo/portfile.cmake:47]: > > It matches a single line.
> 
> `REGEX` matches only once, but `.*` also matches line breaks. So the proposed change matches multiple lines.

Yeah right 🤦

**Comment by ras0219-msft** [comment]
If https://github.com/microsoft/vcpkg-tool/pull/172#discussion_r776026107 may or may not be added, it seems like the regex is probably necessary.
```
"\n#define ARMA_AUX_LIBS [^\\n]*\n"
```
should avoid matching multiple lines iiuc.

**Comment by dg0yt** [comment]
@ras0219-msft `vcpkg_replace_string` doesn't support a regex.

**Comment by ras0219-msft** [comment]
LGTM, thanks!

>`vcpkg_replace_string` doesn't support a regex.

Sorry for the brevity, I meant to say that we should switch back to the `string(REGEX REPLACE)` approach with the regex "blah"

**Pull request merged**